src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
 
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('triangle cannot exist');
+    }
 
-export class Circle {
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('triangle cannot exist');
+    }
+  }
```

Comment:
- the error messages must be more descriptive
- as you have two errors to throw - it is definitely better to create some separate method like `validateTriangle` or sth